### PR TITLE
Nanoscale date time for elastic search

### PR DESCRIPTION
**Pull request by aliotta**
## Description

- adds an elasticsearch index template which ensures that the date_nano field is of type date_nanos
- adds a date_nano field to our fluentd records

Related PRs:
houston: https://github.com/astronomer/houston-api/pull/1253
ui: https://github.com/astronomer/astro-ui/pull/1505

## Related Issues

Related astronomer/issues#385

## Testing

I tested in aws with all three components and it works as expected.
 
## Merging

0.31.0

### CODE REVIEW

File: charts/fluentd/templates/fluentd-index-template-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: astronomer-elasticsearch-index-template
+  labels:
```
Comment (pgvishnuram) [inline charts/fluentd/templates/fluentd-index-template-configmap.yaml:5]: this is still hardcoded

File: charts/fluentd/templates/fluentd-daemonset.yaml [hunk 2]
```diff
@@ -130,3 +133,6 @@ spec:
       - name: config-volume-{{ template "fluentd.fullname" . }}
         configMap:
           name: {{ template "fluentd.fullname" . }}
+      - name: astronomer-elasticsearch-index-template
+        configMap:
+          name: astronomer-elasticsearch-index-template
```
Comment (pgvishnuram) [inline charts/fluentd/templates/fluentd-daemonset.yaml:138]: this is still hardcoded

File: charts/fluentd/templates/fluentd-index-template-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    tier: logging
+    component: {{ template "fluentd.name" . }}
+    chart: {{ template "fluentd.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+data:
+  index_template.json: |-
+    {
+      "index_patterns": [
+          "fluent*"
```
Comment (pgvishnuram) [inline charts/fluentd/templates/fluentd-index-template-configmap.yaml:15]: its flentd.*

File: charts/fluentd/templates/fluentd-daemonset.yaml (hunk 2 above)
Comment (aliotta) [reply charts/fluentd/templates/fluentd-daemonset.yaml]: done

File: charts/fluentd/templates/fluentd-index-template-configmap.yaml (hunk 3 above)
Comment (aliotta) [reply charts/fluentd/templates/fluentd-index-template-configmap.yaml]: done

File: charts/fluentd/templates/fluentd-index-template-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v1
```
Comment (pgvishnuram) [inline charts/fluentd/templates/fluentd-index-template-configmap.yaml:1]: Add header for better helm render

Comment (aliotta) [reply charts/fluentd/templates/fluentd-index-template-configmap.yaml:4]: done

**Pull request merged**